### PR TITLE
add docker controller

### PR DESCRIPTION
**Pull request by miltontony**

### CODE REVIEW

**Comment by miltontony** [comment]
@smn finally, the review I've been building up towards \o/
I'm adding a new controller type (basic docker controller). I forsee other controllers to follow (vumi, unicore, etc). But I was also hoping to make controllers outside of the mc2 codebase so that we can `pip install` them whilst keeping the mc2 codebase fairly clean.

File: controllers/docker/test_docker_controller.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import pytest
+import responses
```
Comment (smn) [inline controllers/docker/test_docker_controller.py]: What's the latest one? This file or `tests/test_docker_controller.py`? Perhaps you committed both due to a move to a `tests/` folder?

**Comment by smn** [comment]
:+1: in general, just a question about the seemingly duplicated controller test?

**Comment by miltontony** [comment]
yes, twas a duplicate :)

**Comment by smn** [comment]
:+1:

**Pull request merged**